lib/chef/config.rb

```diff
@@ -497,7 +497,8 @@ def self.path_accessible?(path)
     default(:syntax_check_cache_path) { cache_options[:path] }
 
     # Deprecated:
-    default(:cache_options) { { :path => PathHelper.join(file_cache_path, "checksums") } }
+    # Move this to the default value of syntax_cache_path when this is removed.
+    default(:cache_options) { { :path => PathHelper.join(config_dir, "syntaxcache") } }
```

Comment:
I think it might if client.rb doesn't contain an explicit value for that path.  From what I could tell, the new location was ~/.chef/syntaxcache.  It didn't seem to fall under the ~/.chef/cache hierarchy.  I initally looked at using join(cache_path, "syntaxcache") but that was adding a 'local-mode-cache' path element to everything when run in local mode.  If this isn't the right place to land the cache, let me know and I'll change it.  I just need it so that knife and chef continue to work when a bare bones starter kit is downloaded and this starter kit doesn't contain any cache_options or syntax_cache entries.